gix-pack/src/index/access.rs

```diff
@@ -49,9 +49,9 @@ impl index::File {
         let pack64_offset = self.offset_pack_offset64_v2();
         match self.version {
             index::Version::V2 => izip!(
-                self.data[V2_HEADER_SIZE..].chunks(self.hash_len),
```

Comment:
Let's leave this at `chunks` as well, it should panic if the count is off (which it will).

Reply:
This could truncate `izip!` too, so maybe an explicit check of all 3 lengths would be better?.

Reply:
I've added length check, which helps with zip too.

Reply:
Thank you, very readable and a much stronger assertion.